data/changelog/ppxlib/2024-07-25-ppxlib-0.33.0.md

```diff
@@ -0,0 +1,111 @@
+
+  - New functions `Ast_builder.{e,p}list_tail` that take an extra tail
+  expression/pattern argument parameter compared to `Ast_builder.{e,p}list`, so
+  they can build ASTs like `a :: b :: c` instead of only `[ a; b ]`.
+  (#498, #502, @v-gb, @NathanReb)
+
+  - Fix `Longident.parse` so it also handles indexing operators such as
+  `.!()`, `.%(;..)<-`, or `Vec.(.%())` (#494, @octachron)
+
+  - Add a `special_function'` variant which directly takes a `Longident.t`
+  argument to avoid the issue that `Longident.t` cover distinct syntaxic classes
+  which cannot be easily parsed by a common parser (#496, @octachron).
+
+  - Keep location ranges consistent when migrating `Pexp_function` nodes from 5.2+
+  to older versions (#504, @jchavarri)
+
+  - Fix `-locations-check` behaviour so it is no longer required to pass `-check` and
+  can enable location checks. (#506, @NathanReb)
+---
+
+The Ppxlib dev team is happy to announce the release of `ppxlib.0.33.0`.
+
+The main feature of this release is a series of improvement to flags controlling
+the silencing of unused value/module/type warnings.
+The `ppxlib` driver generates warning silencing items to prevent `[@@deriving ...]`
+generated code to trigger unused code warnings. Three warnings are disabled that
+way:
+- Warning 32: unused value
+- Warning 60: unused module
+- Warning 34: unused type
+The first two are disabled for values and modules generated by the deriver while
+the third is disabled for the types in the type declaration to which the
+`[@@deriving ...]` attribute is attached.
+This feature was added a long time ago to avoid manually disabling those
+warnings when working with derivers that generate a set of values and modules
+only to use a subset of those. Alternatively, the unused type warning silencing
+was added to allow defining an alias type only to be consumed by a deriver (e.g.,
+``type error = [`Not_found | `Invalid_arg] [@@deriving to_string]``).
+We since then believe that we should not disable warnings lightly and this
+behaviour makes it difficult to find and remove deadcode. The right approach in
+those situations should be to fix the PPX derivers so that they are more
+configurable and can be used without triggering such warnings.
+We will start to move toward removing this feature, but since it is still useful in
+some places, we came up with a plan to do this iteratively.
+In `ppxlib.0.31.0` we added the `-unused-code-warnings` driver flag and the
+`?unused_code_warnings` `Deriving.V2.make` optional argument to control whether
+to silence Warnings 32 and 60. When both are set to `true`, by the user and the
+deriver authors, the warnings are not silenced. As of `ppxlib.0.33.0`, these also
+control the silencing of Warning 34 (unused type). The `-unused-code-warnings`
+flags can now be passed `force` to disable warnings silencing, regardless of the
+derivers opting in. This allows users to test whether their codebase and their
```

Comment:
```suggestion
control the silencing of Warning 34 (unused type). `force` can now be applied to the `-unused-code-warnings`
flags in order to disable warnings silencing, regardless of the
derivers opting in. This allows users to test whether their codebase and their
```
Even if you don't like this suggestion, should it be "applied" or "passed" -- same with li.100

Reply:
I like the suggestion! It should be `passed` here. I'll change it here and at line 100!